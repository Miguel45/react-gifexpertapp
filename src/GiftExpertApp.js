import React from 'react'
import { AddCategory } from './components/AddCategory'
import { GiftGrid } from './components/GiftGrid'

export const GiftExpertApp = ({ defaultCategories = []}) => {

    const [categories, setCategories] = React.useState(defaultCategories);

    return (
        <>

           <h2>GiftExpertApp</h2> 
           <AddCategory setCategories={setCategories}/>
           <hr/>

           <ol>
               {categories.map( ( e ) => <GiftGrid key={ e } category={ e } />) } 
           </ol>
        </>
    )
}
