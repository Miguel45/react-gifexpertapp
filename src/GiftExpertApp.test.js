import React from 'react' 
import { GiftExpertApp } from './GiftExpertApp'
import { shallow } from 'enzyme'

const categories = ['One punch', 'Hunter X Hunter']

describe( 'Pruebas en el archivo GiftExpertApp.js', () => {

    let wrapper = shallow( <GiftExpertApp defaultCategories={categories}/> );

    beforeEach( () => {
        wrapper = shallow(<GiftExpertApp defaultCategories={categories}/>);
    })

    //Enzyme
    test( 'Debe mostrarse bien el componente en un snapshot', () => {

        expect( wrapper ).toMatchSnapshot();

        expect( wrapper.find('GiftGrid').length ).toBe(categories.length)

    })

})