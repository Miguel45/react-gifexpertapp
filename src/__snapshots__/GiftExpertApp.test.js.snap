// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo GiftExpertApp.js Debe mostrarse bien el componente en un snapshot 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GiftGrid
      category="One punch"
      key="One punch"
    />
    <GiftGrid
      category="Hunter X Hunter"
      key="Hunter X Hunter"
    />
  </ol>
</Fragment>
`;
